FrontEnd/src/ProfilePage/UserProfile.jsx

```diff
@@ -19,38 +18,69 @@ function UserProfilePage({ userUID, userData }) {
   const [userPosts, setUserPosts] = useState([]);
   const [isCreatePostModalShown, setIsCreatePostModalShown] = useState(false);
   const [isFormValid, setIsFormValid] = useState(false);
-  const [serviceCategories, setSelectedCategories] = useState([]);
+  const [serviceCategories, setServiceCategories] = useState([]);
   const [availableCategories] = useState(CATEGORIES);
-  const [isSignOutDropdownVisible, setisSignOutDropdownVisible] =
+  const [newCategory, setNewCategory] = useState("");
+  const [searchLocationTerm, setSearchLocationTerm] = useState("");
+  const [suggestions, setSuggestions] = useState([]);
+  const [selectedLocations, setSelectedLocations] = useState([]);
+  const [isSignOutDropdownVisible, setIsSignOutDropdownVisible] =
     useState(false);
-  const [isDeletePostDropdownVisible, setisDeletePostDropdownVisible] =
+  const [isDeletePostDropdownVisible, setIsDeletePostDropdownVisible] =
     useState(false);
```

Comment:
```suggestion
    useState(null);
```

Reply:
@pearlor I'm a bit confused, why switch a boolean default to `null`?

Reply:
Is there a reason I shouldn't use false?

Reply:
Yeah, you want the types to be consistent. Since this is a useState containing the index of the post to be deleted, we should use either the index or null.